pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,309 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Code relevant for performing measurements on a qutrit mixed state.
+"""
+
+from typing import Callable
+from string import ascii_letters as alphabet
+from pennylane import math
+from pennylane.ops import Sum, Hamiltonian
+from pennylane.measurements import (
+    StateMeasurement,
+    MeasurementProcess,
+    ExpectationMP,
+    StateMP,
+    ProbabilityMP,
+    VarianceMP,
+)
+from pennylane.operation import Observable
+from pennylane.typing import TensorLike
+
+from .utils import (
+    get_einsum_mapping,
+    resquare_state,
+    get_num_wires,
+    get_new_state_einsum_indices,
+    QUDIT_DIM,
+)
+from .apply_operation import apply_operation
+
+alphabet_array = math.asarray(list(alphabet))
+
+
+def apply_observable_einsum(obs: Observable, state, is_state_batched: bool = False):
+    r"""Applies an observable to a density matrix rho, giving obs@state
+
+    Args:
+        obs (Operator): Operator to apply to the quantum state
+        state (array[complex]): Input quantum state
+        is_state_batched (bool): Boolean representing whether the state is batched or not
+
+    Returns:
+        TensorLike: the result of obs@state
+    """
+
+    def map_indices(
+        state_indices, row_indices, new_row_indices, **kwargs
+    ):  # pylint: disable=unused-argument
+        """Map indices to wires
+        Args:
+            state_indices (str): Indices that are summed
+            row_indices (str): Indices that must be replaced with sums
+            new_row_indices (str): Tensor indices of the state
+            **kwargs (dict): Stores indices calculated in `get_einsum_mapping`
+
+        Returns:
+            String of einsum indices to complete einsum calculations
+        """
+        op_1_indices = f"{new_row_indices}{row_indices}"
+
+        new_state_indices = get_new_state_einsum_indices(
+            old_indices=row_indices,
+            new_indices=new_row_indices,
+            state_indices=state_indices,
+        )
+
+        return f"{op_1_indices},...{state_indices}->...{new_state_indices}"
+
+    num_ch_wires = len(obs.wires)
+    einsum_indices = get_einsum_mapping(obs, state, map_indices, is_state_batched)
+    obs_mat = obs.matrix()
+    obs_shape = [QUDIT_DIM] * num_ch_wires * 2
+    obs_mat = math.cast(math.reshape(obs_mat, obs_shape), complex)
+    return math.einsum(einsum_indices, obs_mat, state)
+
+
+def trace_method(
```

Comment:
```suggestion
def calculate_expval(
```

Recommend using a consistent naming scheme for the different measurement function option (I made a similar suggestion for `reduce_dm` as well). It also makes it more clear what it is doing here.